hyperapp.d.ts

```diff
@@ -0,0 +1,68 @@
+export as namespace hyperapp
+
+/** @namespace [Virtual DOM] */
+
+/** The Virtual DOM representation of an Element
+ *
+ * @memberOf [Virtual DOM]
+*/
+export interface VirtualNode<Data> {
+  tag: string
+  data: Data
+  children: VirtualNodeChild<{} | null>[]
+}
+
+/** In the Virtual DOM a Child could be either a Virtual Node or a string
+ *
+ * @memberOf [Virtual DOM]
+*/
+export type VirtualNodeChild<Data> = VirtualNode<Data> | string
+
+/** A Component is a function that return a custom Virtual Node
+ *
+ * @memberOf [Virtual DOM]
+*/
+export interface Component<Data> {
+  (data?: Data, ...children: VirtualNodeChild<{} | null>[]): VirtualNode<Data>
+}
+
+/** The soft way to create a Virtual Node
+ * @param tag       Either a tag name e.g. 'div', 'svg', etc. Or a Component function
+ * @param data      Any valid HTML atributes, events, styles, lifecycle events, and meta data
+ * @param children  The children of the VirtualNode
+ *
+ * @memberOf [Virtual DOM]
+*/
+export function h<Data>(
+  tag: Component<Data> | string,
+  data?: Data,
+  children?:
+    | VirtualNodeChild<{} | null>[]
+    | VirtualNodeChild<{} | null>
+    | number
+): VirtualNode<Data>
+
+/** @namespace [Application] */
+
+export interface Emit<Data, Events> {
```

Comment:
Yup! You can specify a default type, which makes that generic param optional. For example:
```typescript
interface SomeInterface<T, U=any>
```